pkg/build/apis/build/validation/validation.go

```diff
@@ -194,6 +195,22 @@ func validateSource(input *buildapi.BuildSource, isCustomStrategy, isDockerStrat
 		for i, image := range input.Images {
 			allErrs = append(allErrs, validateImageSource(image, fldPath.Child("images").Index(i))...)
 		}
+		var set sets.String
+		for i, image := range input.Images {
+			for j, name := range image.As {
+				if len(name) == 0 {
```

Comment:
probably worth a comment explaining that you don't record an error here because validateImageSource already did.